packages/contracts/contracts/Artonomous.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity ^0.4.24;
+
+import "./ArtonomousStaking.sol";
+import "./ArtonomousArtPieceToken.sol";
+
+contract Artonomous {
+
+    event ArtonomousAuctionStarted(uint indexed blockNumber, string generatorHashUsed);
+    event ArtonomousArtClaimed(address indexed claimant, uint indexed blockNumber);
+
+    struct Auction {
+        uint blockNumber;
+        uint endTime;
+    }
+
+    ArtonomousStaking public artonomousStaking;
+    ArtonomousArtPieceToken public pieceToken;
+    uint public AUCTION_LENGTH = 86400; // 24 hours
+    Auction currentAuction;
+
+    constructor(address stakingAddr) public {
+        artonomousStaking = ArtonomousStaking(stakingAddr);
+        pieceToken = new ArtonomousArtPieceToken("artonomous-token", "ARTO");
+        startAuction();
+    }
+
+    function startAuction() internal {
+        require(currentAuction.blockNumber == 0);
```

Comment:
Perhaps instead of this logic we could assert `pieceToken.ownerOf(currentAuction.blockNumber) != this`. That would allow us to avoid creating/deleting the `currentAuction` variable, which is more expensive than mutating its value (although not sure exactly how the math works out with the gas refund).

So essentially the starting condition for a new auction is that the last piece is no longer owned by Artonomous.